server/src/main/java/org/elasticsearch/action/admin/indices/rollover/Condition.java

```diff
@@ -62,6 +48,23 @@ boolean includedInVersion(Version version) {
         return true;
     }
 
+    @Override
+    public final boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Condition<?> condition = (Condition<?>) o;
+        return Objects.equals(name, condition.name);
```

Comment:
yes. it was not ignored before but it's counter intuitive. Json doesn't allow for duplicated keys with same name, which is the only way you could end up with two conditions with same name. These conditions end up in a set, which is why I think it is important to properly implement equals and hashcode although before it was not.

Reply:
I see, but it feels a bit strange. Maybe the way it is used in the set is a bit off then? Probably outside the scope of this PR though, but can you add this explanation as a javadoc comment to the equals method?

Reply:
It is within the scope of this PR as I introduced these methods as part of this PR :) I am trying to think of how to improve this and make it less strange. Maybe having a set would not not be that important if things only came from the REST layer, but given that we are setting it also through transport client and high-level REST client, I think we need to make sure that a condition can only be set once. Would you prefer to have a `Map<String, Condition>` where the key is the name, rather than a `Set` and implement equals/hashcode this way?

Reply:
+1, that would probably make things clearer. The set might not have been the best choice given we only want to assure each condition type is set once, as far as I understand it.

Reply:
Done.